### PR TITLE
KFSPTS-25323 upgrade to the 4/13/2022 version of financials

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

**Comment by db483** [comment]
retest this please

**Comment by db483** [comment]
retest this please

**Review by cah292 — COMMENTED** [review]
I'm still reviewing the changes, but I did want to bring up one item in the meantime.

File: src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java [hunk 1]
```diff
@@ -180,250 +180,179 @@ public void templateVendor(VendorDetail vendor, VendorAddress vendorAddress) {
...
         } else {
-            this.getDvPayeeDetail().setDisbVchrPayeeEmployeeCode(false);
+            getDvPayeeDetail().setDisbVchrPayeeEmployeeCode(false);
         }
 
         // I'm assuming that if a tax id type code other than 'TAX' is present, then the employee must be foreign
-        for (String externalIdentifierTypeCode : employee.getExternalIdentifiers().keySet()) {
+        for (final String externalIdentifierTypeCode : employee.getExternalIdentifiers().keySet()) {
             if (KimConstants.PersonExternalIdentifierTypes.TAX.equals(externalIdentifierTypeCode)) {
                 this.getDvPayeeDetail().setDisbVchrNonresidentPaymentCode(false);
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java:220]: Thank you for catching this. I will clean this up.

[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java]: I made updates to remove references to `this` that are not needed. For `final` local variables, did you notice some instances that were updated in base code but not in our extension? Or is this s suggestion to update all local variables to `final`? Thank you

[the remark this answers is not in the document]
Comment (cah292) [reply src/main/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocument.java]: Thanks for the cleanup. I had noticed some method parameters that were missing the `final` keyword, but I don't recall whether they were on CU-specific methods or overridden methods.

I'm fine with deferring the rest of the `this` and `final` cleanup to future stories, especially if we end up deciding to add such changes to our regular coding standards. I'll add a bookclub agenda item for discussing with the team.

**Review by cah292 — CHANGES_REQUESTED** [review]
Changes look good overall, but I noticed a few more items to be addressed:

1. For consistency with base code, the method `CuElectronicInvoiceHelperServiceImpl.createPaymentRequest()` may need to be adjusted to use a Spring-injected PersonService instead of calling SpringContext directly. I realize that the `personService` field is a private one, so you may need to add a related field to the subclass and override the setter if you want to avoid introducing an overlay. We should also report this to KualiCo if you think the field's visibility should be increased.
2. The method `CuVendorRuleBase.validateParentVendorTaxNumber()` should add the "active" flag to the appropriate criteria, for consistency with the changes in base code.

**Comment by db483** [comment]
Thank you for your review Chad. I made the requested changes and reported to KualiCo the field visibility increase: KFSPTS-27415.

**Comment by cah292** [comment]
@db483,

You mentioned that you made the requested changes, but in this PR I'm not seeing any new changes to the files I had mentioned. Can you please confirm if your latest changes have been committed yet? Thanks.

**Comment by db483** [comment]
It looks like I forgot to push the changes. they should be available now. Thank you.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass, my concerns have been addressed and the related PRs are also ready. Merging.

**Pull request merged**